linera-base/src/identifiers.rs

```diff
@@ -188,6 +307,57 @@ impl<A> Clone for SessionId<A> {
 
 impl<A> Copy for SessionId<A> {}
 
+impl<A: std::cmp::PartialEq> PartialEq for SessionId<A> {
+    fn eq(&self, other: &Self) -> bool {
+        let SessionId {
+            application_id,
+            index,
+        } = other;
+        self.application_id == *application_id && self.index == *index
+    }
+}
+
+impl<A: std::cmp::PartialEq> Eq for SessionId<A> {}
+
+impl<A: std::cmp::PartialEq> PartialOrd for SessionId<A> {
```

Comment:
Looks like you can remove all these `std:cmp::`. Also, (even if the compiler doesn't require it?), in this case, I would expect the trait bound to be the same as the implemented trait (here `impl<A: PartialOrd>`).